src/toil_scripts/tools/preprocessing.py

```diff
@@ -95,12 +150,88 @@ def run_picard_create_sequence_dictionary(job, ref_id):
     work_dir = job.fileStore.getLocalTempDir()
     job.fileStore.readGlobalFile(ref_id, os.path.join(work_dir, 'ref.fasta'))
     command = ['CreateSequenceDictionary', 'R=ref.fasta', 'O=ref.dict']
-    docker_call(work_dir=work_dir, parameters=command,
+    outputs = {'ref.dict': None}
+    docker_call(work_dir=work_dir,
+                parameters=command,
+                outputs=outputs,
+                env={'_JAVA_OPTIONS':'-Djava.io.tmpdir=/data/ -Xmx{}'.format(xmx)},
                 tool='quay.io/ucsc_cgl/picardtools:1.95--dd5ac549b95eb3e5d166a5e310417ef13651994e')
     return job.fileStore.writeGlobalFile(os.path.join(work_dir, 'ref.dict'))
 
 
-def run_gatk_preprocessing(job, cores, bam, bai, ref, ref_dict, fai, phase, mills, dbsnp, mem='10G', unsafe=False):
+def picard_sort_sam(job, bam_id, xmx=10737418240):
```

Comment:
OOC, why do we have both samtools and picard sort? Surely we're just using one? And if we are just using one, it should be neither samtools nor picard, it should be sambamba ;).